pkg/domain/infra/abi/quadlet.go

```diff
@@ -0,0 +1,764 @@
+	// We can find them with LoadState, as we do above.
+	// Handle it anyways because it's easy enough to do.
+	for _, report := range partialReports {
+		report.Status = "Not loaded"
+		reports = append(reports, &report)
+	}
+
+	finalReports := make([]*entities.ListQuadlet, 0, len(reports))
+	for _, report := range reports {
+		include := true
+		for _, filterFunc := range filterFuncs {
+			include = filterFunc(report)
+		}
+		if include {
+			finalReports = append(finalReports, report)
+		}
+	}
+
+	return finalReports, nil
+}
+
+// Retrieve path to a Quadlet file given full name including extension
+func getQuadletPathByName(name string) (string, error) {
+	// Check if we were given a valid extension
+	if !systemdquadlet.IsExtSupported(name) {
+		return "", fmt.Errorf("%q is not a supported quadlet file type", filepath.Ext(name))
+	}
+
+	quadletDirs := systemdquadlet.GetUnitDirs(rootless.IsRootless())
+	for _, dir := range quadletDirs {
+		testPath := filepath.Join(dir, name)
+		if _, err := os.Stat(testPath); err != nil {
+			if !errors.Is(err, fs.ErrNotExist) {
+				return "", fmt.Errorf("cannot stat quadlet at path %q: %w", testPath, err)
+			}
+			continue
+		}
+		return testPath, nil
+	}
+	return "", fmt.Errorf("could not locate quadlet %q in any supported quadlet directory", name)
+}
+
+func (ic *ContainerEngine) QuadletPrint(ctx context.Context, quadlet string) (string, error) {
+	quadletPath, err := getQuadletPathByName(quadlet)
+	if err != nil {
+		return "", err
+	}
+
+	contents, err := os.ReadFile(quadletPath)
+	if err != nil {
+		return "", fmt.Errorf("reading quadlet %q contents: %w", quadletPath, err)
+	}
+
+	return string(contents), nil
+}
+
+// QuadletRemove removes one or more Quadlet files from the system.
+// It handles both individual Quadlet files and Quadlet applications (collections of related files).
+// The function performs the following operations:
+//   - Validates that specified Quadlet files exist
+//   - Checks if associated systemd services are running
+//   - Optionally stops running services (if force option is set)
+//   - Removes the Quadlet files and any associated asset files
+//   - Reloads systemd daemon if necessary
+//
+// Parameters:
+//   - ctx: Context for cancellation and timeout handling
+//   - quadlets: List of Quadlet names to remove, or empty slice if options.All is true
+//   - options: Configuration options including Force, Ignore, and All flags
+//
+// Returns:
+//   - QuadletRemoveReport: Contains lists of successfully removed quadlets and any errors encountered
+//   - error: Non-nil if a fatal error occurred that prevented the operation from completing
+//
+// The function supports:
+//   - Force removal of running Quadlet services
+//   - Ignoring missing Quadlet files (with Ignore option)
+//   - Removing all installed Quadlets (with All option)
+//   - Handling Quadlet applications (.app files containing multiple related Quadlets)
```

Comment:
Personally not sure how I feel about all this AI text, I mean it isn't exactly wrong but it is just such a wall of text with very little actual important information.